### PR TITLE
Convenience operators for typed pub sub from streams

### PR DESCRIPTION
**Pull request by johanandren**
Refs #31037

### CODE REVIEW

**Comment by johanandren** [comment]
Probably needs something in the reference docs as well.

File: akka-stream-typed/src/test/scala/akka/stream/typed/scaladsl/PubSubSpec.scala [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import akka.actor.typed.internal.pubsub.TopicImpl
+import akka.actor.typed.pubsub.Topic
+import akka.stream.OverflowStrategy
+import akka.stream.scaladsl.Source
+import akka.stream.testkit.scaladsl.TestSink
+import org.scalatest.wordspec.AnyWordSpecLike
+
+class PubSubSpec extends ScalaTestWithActorTestKit with AnyWordSpecLike {
+
+  implicit def sys = testKit.system.classicSystem
```
Comment (patriknw) [inline akka-stream-typed/src/test/scala/akka/stream/typed/scaladsl/PubSubSpec.scala:20]: can't it use the Typed ActorSystem?

**Review by patriknw — APPROVED** [review]
LGTM

File: akka-stream-typed/src/test/scala/akka/stream/typed/scaladsl/PubSubSpec.scala (hunk 1 above)
Comment (johanandren) [reply akka-stream-typed/src/test/scala/akka/stream/typed/scaladsl/PubSubSpec.scala:20]: I didn't think so but turns out there is an `TestSink.apply()` which takes `ClassicActorSystemProvider`. I'll update.

**Review by octonato — APPROVED** [review]
LGTM

**Pull request merged**